### DISCUSSION TITLE
How would you call this library from python

### DISCUSSION THREAD

**Discussion by alpha027** [TheLartians/Greeter#173] [Q&A]
So using the same architecture, how would one wrap the C++ code to Python and be able to use it and call methods and functions from Python?

**Comment by cauliyang** [comment]
Using https://github.com/pybind/pybind11 is a good idea to bind your C++ code to Python. 
It provides full examples of different build systems. 
For Cmake, you can reference here, https://github.com/pybind/cmake_example. 
Hence, it is not hard to add a shared library that Python can use. 
By the way, https://github.com/RosettaCommons/binder is a great tool to reduce boilerplate binding code.
